security/pkg/pki/util/verify_cert.go

```diff
@@ -58,50 +59,53 @@ func VerifyCertificate(privPem []byte, certChainPem []byte, rootCertPem []byte,
 		return err
 	}
 
-	host := expectedFields.Host
-	san := host
-	// uri scheme is currently not supported in go VerifyOptions. We verify
-	// this uri at the end as a special case.
-	if strings.HasPrefix(host, "spiffe") {
-		san = ""
-	}
 	opts := x509.VerifyOptions{
-		DNSName:       san,
 		Intermediates: intermediates,
 		Roots:         roots,
 	}
+	host := ""
```

Comment:
nit: looks like most of these changes are just adding nil checks. IF not, please ensure that adequate changes are made to the unit tests

Reply:
yes, all of these changes are just adding nil checks.